test/metric/metric_value_query.go

```diff
@@ -61,7 +62,7 @@ type MetricValueFetcher interface {
 	isApplicable(metricName string) bool
 
 	// getMetricSpecificDimensions returns the dimensions that needs to be scraped by each plugin
-	getMetricSpecificDimensions() []types.Dimension
+	getMetricSpecificDimensions(metricName string) []types.Dimension
```

Comment:
I was a bit confused at the beginning when seeing most of the tests we don't even pass in a string to `getMetricSpecificDimensions`, but I see the reason from the prometheus test. I guess another way might be just implement a default `getMetricSpecificDimensions` in prometheus test without calling it and create another helper function to return the dimensions based on the metric name. Not blocking this though

Reply:
I see how that can be confusing. But thought more plugins could have this usecase in the future so thought its better to update the interface method itself.